\documentclass{article}
\begin{document}
	\paragraph*{Question 340} $\boxed{\text{Difficulty: Easy}} $			
	
	According to the C++17 standard, what is the output of this program?
	
	\begin{lstlisting}[language=C++]  		
#include <future>
#include <iostream>

int main()
{
	try {
		std::promise<int> p;
		std::future<int> f1 = p.get_future();
		std::future<int> f2 = p.get_future();
		p.set_value(1);
		std::cout << f1.get() << f2.get();
	} catch(const std::exception& e)
	{
		std::cout << 2;
	}
}
	\end{lstlisting}
	
	\paragraph*{答案和解析} $\boxed{\text{程序正常运行，输出为：2}} $
	
	\href{https://timsong-cpp.github.io/cppwp/n4659/futures.promise#12}{futures.promise\#12}
	\begin{lightgrayleftbar}
		future<R> get\_future();
		
		Returns: A future<R> object with the same shared state as *this.
		
		Throws: future\_­error if *this has no shared state or if 
	\textbf{get\_­future has already been called on a promise with the same shared state as *this.}
	\end{lightgrayleftbar}
\end{document}
